blog/content/post/2024-04-12-odroid-m1-secure-boot.md

```diff
@@ -0,0 +1,649 @@
+---
+title: Enabling Secure Boot on ODROID M1 (RK3568B)
+abstract: 'This blog post describes how to enable Secure Boot/Mode on
+ODROID-M1(RK3568B). Read how to write hash of public key to OTP memory, how to
+sign loader and how to build signed U-Boot with enabled signature verification.'
+cover: /covers/rockchip-logo.jpg
+author: michal.iwanicki
+layout: post
+published: true
+date: 2024-04-12
+archives: "2024"
+
+tags:
+  - bootloader
+  - secure-boot
+  - rockchip
+  - u-boot
+categories:
+  - Firmware
+  - Security
+
+---
+
+## Introduction
+
+Recently, I was tasked with implementing Secure Boot on the ODROID M1
+(RK3568B).
+In this post, I will describe how I managed to store the hash of the public key
+in the OTP memory and enabled verification of the pre-loader's signature
+(Rockchip TPL + U-Boot SPL) as well as the verification of the U-Boot's
+configuration by SPL which includes hashes of all images contained in the FIT
+image.
+
+## Enabling Secure Mode
+
+Enabling Secure Mode is based on storing the hash of the public key in
+the OTP memory and enabling verification of the pre-loader’s signature.
+After correctly executing this step, the RK3568B SoC will check if the hash
+of the public key that is contained in the booted image matches the hash
```

Comment:
Changed in [9605d08](https://github.com/3mdeb/news-and-ideas/pull/593/commits/9605d0863b5fa9a1961c20d7812398b0388ca4f2), rewrote it a little and added a diagram.
Even on your slide (18) there is information that it only saves hash to OTP/EFUSE.

Reply:
@TomaszAIR And I'm not sure if I can use/store this diagram from Rockchip. I took it from `Rockchip Secure Boot Application Note_v1.2_20160202.pdf` that's inside this [zip](https://github.com/JeffyCN/mirrors/blob/tools/linux/Linux_SecureBoot/SecureBootTool_v1.85_foruser.zip). There are also interesting diagrams in [Rockchip Secure Boot Application Note](http://resource.milesight-iot.com/files/Rockchip-Secure-Boot-Application-Note-V1.9.pdf#header-n150).
Or should I make one myself?

Reply:
@m-iwanicki Since the linked document is classified as "Publicity," it indicates that it is likely meant to be shared publicly, but this doesn't always automatically grant permission for unrestricted use. What we can do is:
1. Create our own simple diagram
2. Add the annotation with the source

Reply:
@m-iwanicki, have you addressed this?

Reply:
I guess you are right. Just checked to make sure that comments are also added in `html` page.